lib/bsdlib/pm.yml

```diff
@@ -0,0 +1,6 @@
+#include <autoconf.h>
```

Comment:
I kind of dislike mixing C-preprocessing into yml files, as it mixes two different principles.

Although I see this was already introduced here: https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/705
I think we should consider if this could be done in a cleaner way, that would remove this directive.

Reply:
We do this to get a hold of the kconfig values. 

we could use the linker magic to automatically inject the header file to these files, but IMO that is even worse.

Reply:
The code appears to be out of sync with https://github.com/nrfconnect/sdk-zephyr/pull/275
Or maybe both PRs just needs a rebase.

It appears to to me that this file will be copied into a `build/.../generated` folder and used from there.
If that is the case, why not use: `configure_file()`:
https://cmake.org/cmake/help/latest/command/configure_file.html

Then you can write:
```
bsdlib_ram:
  placement: {after: spm_ram}
  size: ${CONFIG_PM_PARTITION_SIZE_BSDLIB_RAM}
  region: ram_primary
```
and CMake will replace `${CONFIG_PM_PARTITION_SIZE_BSDLIB_RAM}` with the correct value for you.
No linker magic or `#include`

Reply:
That sounds a lot better than involving the preprocessor, however we will not be able to get a hold of devicetree symbols with that solution, since those are not easily available in cmake AFAIK?

Reply:
but this is a KConfig symbol. Those are available in CMake.

Reply:
In other `pm.yml` files we use `DT` symbols, so we already need the preprocessing. IMO it would not be good adding the `configure_file` as well.

However, I guess at some point one would need to export DT symbols to cmake (as it was before), then we can go with your solution.

Reply:
> In other pm.yml files we use DT symbols, so we already need the preprocessing.

then keep current proposal. I agree, there should only be one way to do this.